### PR TITLE
simplify initialize API in assignment strategies and contribution

### PR DESCRIPTION
**Pull request by AnthonyBuisset**

### CODE REVIEW

File: contracts/onlydust/marketplace/constants.cairo [hunk 1]
```diff
@@ -0,0 +1,8 @@
+namespace classes {
```
Comment (tdelabro) [inline contracts/onlydust/marketplace/constants.cairo:1]: hash_classes

File: contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo [hunk 2]
```diff
@@ -80,31 +79,18 @@ func strategies{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}
...
     func store_strategy_loop{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
-        calldata_len, calldata: felt*
-    ) -> felt {
-        alloc_locals;
-
-        if (calldata_len == 0) {
-            return (0);
+        strategies_len, strategies: felt*
+    ) {
+        if (strategies_len == 0) {
```
Comment (tdelabro) [inline contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo:84]: Is a composite strategy without any strategy really a valid one?
Should we not revert instead ?

**Review by tdelabro — APPROVED** [review]

File: contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo (hunk 2 above)
Comment (AnthonyBuisset) [reply contracts/onlydust/marketplace/core/assignment_strategies/composite.cairo:84]: This is the stop condition for the loop

File: contracts/onlydust/marketplace/constants.cairo (hunk 1 above)
Comment (AnthonyBuisset) [reply contracts/onlydust/marketplace/constants.cairo]: done

**Pull request merged**